### ISSUE TITLE
Aggregates in ORDER BY are not validated/inspected

### ISSUE THREAD

**Issue by justinj** [MaterializeInc/materialize#3183]
```
materialize=> select a from x order by max(a);;
ERROR:  Internal error: encountered unplanned aggregate function: Function { name: ObjectName([Ident("max")]), args: Args([Identifier([Ident("a")])]), filter: None, over: None, distinct: false }
```

**Comment by justinj** [comment]
Same underlying issue as #2804, I think, which is that expressions in an ORDER BY are not treated the same way as other expressions in a SELECT clause.

**Comment by benesch** [comment]
Just realized this was a duplicate of #828. But more importantly, it's fixed by #3645!
